### ISSUE TITLE
Remove the optional taming of Date

### ISSUE THREAD

**Issue by jfparadis** [Agoric/SES-shim#202]
The `Date` constructor of the start compartment should remain unchanged. Only new compartments get initialized with the tamed Date. New compartments can be endowed with the unsafe Dare.

```js
// in the start compartment, a new compartment is endowed with the unsafe Date.
new Compartment({ Date });
```

FYI @erights

**Comment by jfparadis** [comment]
Side effect: with this approach, the `Date.prototype` is always disabled, and `dateInstance.toLocaleString` alwas returns `NaN`.

@erights, are you ok with this limitation?

**Comment by erights** [comment]
`Date.prototype` itself is just a plain object. Did you mean

 " `Date.prototype.constructor` is always disabled "

?

I have not yet thought much about locales, but need to. What are your thoughts on locales in SES in general? Likewise for timezones?

**Comment by jfparadis** [comment]
No, the constructor isn't an issue, because just with Function, we can create new instances using the safe and unsafe constructors that are appear to be instance of both the safe and unsafe constructors. The taming is not intrusive, and can be avoided by using the untamed Date constructor.

`Date.prototype` is the intrinsic `DatePrototype` and is always the same value across all compartments in a Realm.  When we tame `DatePrototype` to disable all unsafe behavior, we do process the `constructor` property but also the `toLocaleString` property.

I think the proposed solution for `Date.now()` and `Math.random()` fails in regards to `.toLocaleString()` because it makes `.toLocaleString()` always disabled on both safe and unsafe Date and Object: 
https://github.com/Agoric/SES-shim/blob/master/packages/tame-global-date-object/test/tameGlobalDateObject.test.js#L52 

In other words creating a compartment with the Date constructor from the start compartment isn't able to avoid all taming. One approach could be to treat locales globally, across all objects, and maintain an option `noTameLocales` or something like that. The solution is not as clean as we thought (we can't get rid of all options).
 
PS: Locales are discussed here: https://github.com/Agoric/SES-shim/issues/149

**Comment by jfparadis** [comment]
@erights Thoughts?

**Comment by erights** [comment]
Well, the default state should be the safe state. As with the Date constructor, if we can find a way to preserve the full safety of the shared intrinsics but still give the start compartment special powers that make it more compatible, that can be good --- depending on the cost of that way. If the cost is low, then again, as with the Date constructor, we can drop the config option and just make it the normal way. It doesn't need to be observably equivalent, or break no code. It just needs to break less code.

Given all that, I have a horrible kludge to suggest. I don't like it, but it follows the above logic. But we may still judge its cost (in kludginess) too high to be worth it.

The unsafe Date constructor we place on the start compartment is not the original unsafe Date constructor. Rather, it wraps the original unsafe Date constructor, is equally unsafe, but has the additional behavior of initializing the date instances it make with a `toLocaleString` ***own*** property.

Does anyone actually use `toLocaleString` anyway? If we just kill it, how much code are we actually breaking?

**Comment by jfparadis** [comment]
> Does anyone actually use toLocaleString anyway? If we just kill it, how much code are we actually breaking?

Yes, that's the biggest question. You can get a sense of the impact, but it's hard to assess:
https://www.google.com/search?q=toLocaleString+site%3Agithub.com

This why I suggest that we split the problem in two: focus on the date constructor (that resolve "preventing access to time"), and we address all of the locales in a different mode of taming (to resolve a different problem, getting information about the host). This mean we remove the `noTameDate` and add a `noTameLocales`. 

Based on https://github.com/Agoric/SES-shim/issues/149, it seems that the second problem need a more advanced solution anyways.

**Comment by erights** [comment]
Sounds good.

**Comment by kriskowal** [comment]
@erights landed and released changes recently that allow the start compartment’s Date to remain powerful depending on the dateTaming option. All other compartments receive powerless Dates.

I believe this issue is resolved. Thank you!
